### ISSUE TITLE
Add testing framework

### ISSUE THREAD

**Issue by matt-fidd** [BritishYouthBandAssociation/Style#13]
I think that we should start early in our approach to testing.

Adding a testing framework at this stage will prevent regression (albeit there's little to regress), make sure that the css builds as expected and that the builder class can handle erroneous data.

It will also allow us to build a github action that automatically runs tests against pull requests as an extra layer of certainty before approving and merging.

**Comment by rugulous** [comment]
Yeah there's defo improvements that can be made (I've broken it several times) so not a bad shout!

**Comment by matt-fidd** [comment]
The tests I propose are 

General
- Sass sheet builds with no errors 

Builder class
- Can set self.vars 
- Retrieves vars correctly 
- Can build with no vars 
- Builds correctly with overwrites 
- Does something (need to look into an intended action) with erroneous data

**Comment by rugulous** [comment]
@matt-fidd would agree.

Guess the erroneous data could be e.g. `$default: #FFFFFFFFFFFFF` (obviously wrong colour) or `$error: null` etc?

**Comment by matt-fidd** [comment]
That was my thinking, what do reckon the intended action should be? If the value isn't something we expect we throw an error? Or just throw away that value? 

And are we assuming that the only valid type is a 6 digit hex code with `#` at the start? I know there are a lot more options than that but how are we going to check for them?

**Comment by rugulous** [comment]
Tbh I'm not too sure - should we validate here?

Just thinking that a colour can be in many formats (`#FFF`, `#123456`, `#12345678`, `rgb(0,0,0)`, `rgba(3,78,90,0.5)`, probably others ...) so it will be rather difficult to validate just these values.

Then there's validating if we pass other values e.g. font names, sizings, paddings, margins...

Perhaps this method allows any value to be passed, then when `build` is called and the SCSS compiler either builds or throws and error it can be properly picked up?

**Comment by matt-fidd** [comment]
That makes most sense I think, I imagine sass will just throw an error itself if we do something stupid, we can always handle that error gracefully if we wanted or just let it bubble up.

**Comment by rugulous** [comment]
Yep, `build` will throw an error like this if
1. the variable contains something stupid
2. we've messed up the actual scss files

![image](https://user-images.githubusercontent.com/13632306/156469403-ad321197-34c6-4fd2-b892-63f6f91df11c.png)

So does seem like a good place to catch it!

**Comment by matt-fidd** [comment]
Sorry, forgot about this issue.

In that case I propose that we keep the Builder as it is, and allow those errors to bubble up to the calling scope so that they can be handled there.
